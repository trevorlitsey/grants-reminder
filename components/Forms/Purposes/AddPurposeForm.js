import { func } from 'prop-types';

class AddPurposeForm extends React.PureComponent {

	constructor(props) {
		super(props);

		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);

		this.state = {
			newPurpose: '',
		}
	}

	handleChange = (e) => {
		this.setState({ newPurpose: e.target.value })
	}

	handleSubmit = (e) => {
		e.preventDefault();
		const { newPurpose } = this.state;
		if (newPurpose) this.props.addNewPurpose(newPurpose);
		this.setState({ newPurpose: '' })
	}

	render() {
		return (
			<div>
				<form onSubmit={this.handleSubmit}>
					<input
						data-test="new-purpose"
						type="text"
						name="purpose"
						placeholder="New purpose ..."
						value={this.state.newPurpose}
						onChange={this.handleChange}
					/>
					<input
						className="submit"
						type="submit"
						value="submit"
					/>
				</form>

				<style jsx>{`
					
					form {
						margin-top: 8px;
						margin-bottom: 0;
						display: flex;
					}
					
					input {
						margin: 0 2px;
					}
					
					.submit {
						padding: 0 10px;
					}
					
				`}</style>
			</div>
		)
	}
}

AddPurposeForm.propTypes = {
	addNewPurpose: func.isRequired,
}

export default AddPurposeForm;