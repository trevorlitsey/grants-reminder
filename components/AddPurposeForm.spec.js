import React from 'react';
import { shallow } from 'enzyme';

import AddPurposeForm from './AddPurposeForm';

describe('AddPurposeForm', () => {
	it('should render without crashing', () => {
		const wrapper = shallowRenderAddPurposeForm();
		expect(wrapper).toMatchSnapshot();
	})

	it('should update state change on input change', () => {
		const e = {
			target: {
				value: 'what a great new purpose',
			}
		}
		const wrapper = shallowRenderAddPurposeForm();
		const instance = wrapper.instance()
		instance.handleChange(e)
		expect(instance.state.newPurpose).toEqual(e.target.value)
	})

	it('should update state with input value', () => {
		const newPurpose = 'what a great new purpose'
		const wrapper = shallowRenderAddPurposeForm();
		wrapper.setState({ newPurpose })
		const newPurposeInput = wrapper.find('[data-test="new-purpose"]')
		expect(newPurposeInput.getElement().props.value).toEqual(newPurpose)
	})

	it('should should update state on change', () => {
		const instance = shallowRenderAddPurposeForm().instance();

		const newPurpose = 'wow! a new purpose'
		const event = {
			target: {
				value: newPurpose,
			}
		}

		// reflect change
		instance.handleChange(event)
		expect(instance.state.newPurpose).toEqual(newPurpose);

	})

	it('should clear form on sunbmit', () => {
		const instance = shallowRenderAddPurposeForm().instance();

		const event = {
			preventDefault: () => { },
		}

		// reset
		instance.handleSubmit(event);
		expect(instance.state.newPurpose).toEqual('');
	})

})

function shallowRenderAddPurposeForm(props = {}) {
	const propsToUser = {
		addNewPurpose: () => { },
		...props
	}
	return shallow(<AddPurposeForm {...propsToUser} />)
}