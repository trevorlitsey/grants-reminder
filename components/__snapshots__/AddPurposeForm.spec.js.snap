// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPurposeForm should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddPurposeForm
    addNewPurpose={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="jsx-621825576"
          onSubmit={[Function]}
        >
          <input
            className="jsx-621825576"
            data-test="new-purpose"
            name="purpose"
            onChange={[Function]}
            placeholder="New purpose ..."
            type="text"
            value=""
          />
          <input
            className="jsx-621825576 submit"
            type="submit"
            value="submit"
          />
        </form>,
        <JSXStyle
          css="form.jsx-621825576{margin-top:8px;margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}input.jsx-621825576{margin:0 2px;}.submit.jsx-621825576{padding:0 10px;}"
          styleId="621825576"
        />,
      ],
      "className": "jsx-621825576",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="jsx-621825576"
              data-test="new-purpose"
              name="purpose"
              onChange={[Function]}
              placeholder="New purpose ..."
              type="text"
              value=""
            />,
            <input
              className="jsx-621825576 submit"
              type="submit"
              value="submit"
            />,
          ],
          "className": "jsx-621825576",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "jsx-621825576",
              "data-test": "new-purpose",
              "name": "purpose",
              "onChange": [Function],
              "placeholder": "New purpose ...",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "jsx-621825576 submit",
              "type": "submit",
              "value": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": "form.jsx-621825576{margin-top:8px;margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}input.jsx-621825576{margin:0 2px;}.submit.jsx-621825576{padding:0 10px;}",
          "styleId": "621825576",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="jsx-621825576"
            onSubmit={[Function]}
          >
            <input
              className="jsx-621825576"
              data-test="new-purpose"
              name="purpose"
              onChange={[Function]}
              placeholder="New purpose ..."
              type="text"
              value=""
            />
            <input
              className="jsx-621825576 submit"
              type="submit"
              value="submit"
            />
          </form>,
          <JSXStyle
            css="form.jsx-621825576{margin-top:8px;margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}input.jsx-621825576{margin:0 2px;}.submit.jsx-621825576{padding:0 10px;}"
            styleId="621825576"
          />,
        ],
        "className": "jsx-621825576",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="jsx-621825576"
                data-test="new-purpose"
                name="purpose"
                onChange={[Function]}
                placeholder="New purpose ..."
                type="text"
                value=""
              />,
              <input
                className="jsx-621825576 submit"
                type="submit"
                value="submit"
              />,
            ],
            "className": "jsx-621825576",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "jsx-621825576",
                "data-test": "new-purpose",
                "name": "purpose",
                "onChange": [Function],
                "placeholder": "New purpose ...",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "jsx-621825576 submit",
                "type": "submit",
                "value": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": "form.jsx-621825576{margin-top:8px;margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}input.jsx-621825576{margin:0 2px;}.submit.jsx-621825576{padding:0 10px;}",
            "styleId": "621825576",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
